storage/Utils/FileUtils.h

```diff
@@ -54,14 +61,38 @@ namespace storage
 
 	string get_fullname() const { return path + "/" + name; }
 
-    protected:
+    private:
 
-	string path;
+	const string path;
 	string name;
 
+	/**
+	 * Either value of TMPDIR or as a fallback /tmp.
+	 */
+	static string default_path();
+
     };
 
-}
 
+    class TmpMount : public TmpDir
+    {
+
+    public:
+
+	enum Mode { READ_ONLY, READ_WRITE };
```

Comment:
Yes, I can change that.

Since the enum is already visible only in the class namespace I did not consider it nessesary.